ReSharper.FSharp/src/FSharp.Psi.Features/src/Daemon/Analyzers/LambdaAnalyzer.fs

```diff
@@ -95,7 +119,8 @@ type LambdaAnalyzer() =
 
     override x.Run(lambda, _, consumer) =
         let expr = lambda.Expression.IgnoreInnerParens()
-        if not (isApplicable expr) then () else
+        if not (isLambdaApplicable lambda) then () else
```

Comment:
It would peobably be better to do this check only for matched lambdas and not for every one we check here?

Reply:
yes, there was such an idea. But I have not found a nice way to insert this check for every warning type. 
Maybe you have any ideas on how to do it neatly?

Reply:
there was such an idea, but I didn't find a neat way to insert this check for every type of warning.  Maybe you know how to do it neatly?

Reply:
Not something that I completely like but seems better than alternatives: add a function taking a warning, and check if the scope is applicable there. Something like `addHighlighting` maybe.